### PR TITLE
atdgen: update `abstract` type validation to accept all input by default

### PR DESCRIPTION
**Pull request by metanivek**
This PR follows up on #295 and #263 by making the default validation for `abstract` types pass for any input, while still allowing for custom validation.

I used the following definition for testing, which appears to my eyes to be correct and compiles successfully (with a small `util.ml` with always-pass validation definitions).
```
type raw_json = abstract
type raw_json_valid = abstract <ocaml valid="fun _ -> true">

type demo = {
  one: raw_json;
  two: raw_json <ocaml validator="Util.validate_abstract">;
  three: raw_json <ocaml valid="fun _ -> true">;
  four: abstract;
  five: abstract <ocaml validator="Util.validate_abstract">;
  six: abstract <ocaml valid="fun _ -> true">;
  seven: raw_json_valid;
}
```

Feedback very much welcome since this is my first time in this code base!

### PR checklist

- [x] New code has tests to catch future regressions
- [x] Documentation is up-to-date
- [x] `CHANGES.md` is up-to-date

### CODE REVIEW

File: atdgen/src/ov_mapping.ml [hunk 1]
```diff
@@ -307,13 +309,14 @@ let def_of_atd is_shallow (loc, (name, param, an), x) =
...
     match as_abstract x with
     | Some (_, an2) ->
-        Ocaml.get_ocaml_module_and_t Validate name an
-        |> Option.map (fun (types_module, main_module, ext_name) ->
-          let args = List.map (fun s -> Tvar (loc, s)) param in
-          External (loc, name, args,
-                    Ocaml.Repr.External (types_module, main_module, ext_name),
-                    (Validate.get_validator an2, false))
-        )
+        (match Ocaml.get_ocaml_module_and_t Validate name an with
```
Comment (metanivek) [inline atdgen/src/ov_mapping.ml:312]: The best I can tell, there seem to be two notions of "abstract" in the code base. This change allows for `type raw_json = abstract` to work as expected while (I think) still respecting the existing `as_abstract` notion.

**Comment by mjambon** [comment]
Hi! Personally, I haven't done much with validators in a while, so it's not something that was on my mind when I recently extended what we can do with `abstract`. It would be great if you could post an example illustrating the problem you're solving (input, output, expected output). After that, you should add some tests that show that your fix works.

**Comment by mjambon** [comment]
After looking at the tests we have, it looks indeed that we have nothing for testing the validation of `abstract` values. There's the file `test_abstract.atd` which could be reused for testing validation, or you can create a new one just for this. I think it's nice to keep test files focused on one feature rather than putting everything in a giant `test.atd` like we still have (note that you may not be able to add `type foo = abstract` to `test.atd` due to the biniou backend not supporting it). You'll probably have to edit `atdgen/test/dune` to add instructions for running `atdgen -v`.

**Comment by metanivek** [comment]
@mjambon just pushed added/updated tests. If you think it would be good to have tests for the validators (like in my examples), let me know. I'm happy to add that as well.

**Comment by mjambon** [comment]
The tests you added are sufficient, I think. Thank you! I'll let you update the changelog.

**Comment by metanivek** [comment]
@mjambon changelog updated (and title of PR :)

**Comment by mjambon** [comment]
Thank you! Merging and starting a release.

**Pull request merged**